docs/developer-guide/development-process.md

```diff
@@ -0,0 +1,50 @@
+# Development Process
+
+Argo CD is being developed using the waterfall process:
```

Comment:
I think this may be better termed to "Release process and cadence". Also, I'm not sure whether this is actually waterfall. Maybe we should just coin it neutrally, as in "using the following process":

Reply:
Agree with both. I myself did not like `Development Process` but struggled to find a better description. Used `Release process and cadence` instead.